services/gateway_manager/src/main/java/com/futurewei/alcor/gatewaymanager/controller/GatewayController.java

```diff
@@ -0,0 +1,149 @@
+package com.futurewei.alcor.gatewaymanager.controller;
+
+
+import com.futurewei.alcor.common.entity.ResponseId;
+import com.futurewei.alcor.common.utils.RestPreconditionsUtil;
+import com.futurewei.alcor.gatewaymanager.service.GatewayService;
+import com.futurewei.alcor.gatewaymanager.utils.VerifyParameterUtils;
+import com.futurewei.alcor.web.entity.gateway.GatewayEntity;
+import com.futurewei.alcor.web.entity.gateway.GatewayInfo;
+import com.futurewei.alcor.web.entity.gateway.GatewayInfoJson;
+import com.futurewei.alcor.web.entity.gateway.VpcInfoJson;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+@Slf4j
+@RestController
+public class GatewayController {
+
+    @Autowired
+    private GatewayService gatewayService;
+
+    /**
+     * Create a GatewayInfo (for zeta gateway)
+     *
+     * @param projectId
+     */
+    @PostMapping("/project/{projectid}/gatewayinfo")
+    @ResponseStatus(HttpStatus.CREATED)
+    public ResponseId createGatewayInfo(@PathVariable("projectid") String projectId, @RequestBody VpcInfoJson vpcInfoJson) throws Exception {
+        RestPreconditionsUtil.verifyParameterNotNullorEmpty(projectId);
+        VerifyParameterUtils.checkVpcInfo(vpcInfoJson.getVpcInfo());
+        GatewayInfo gatewayInfo = gatewayService.createGatewayInfo(projectId, vpcInfoJson.getVpcInfo());
+        log.info("GatewayInfo created success,GatewayInfo is: {}", gatewayInfo);
+        return new ResponseId(gatewayInfo.getResourceId());
+    }
+
+    /**
+     * Update gatewayInfo (for zeta gateway)
+     *
+     * @param projectId
+     * @param vpcId
+     * @return
+     */
+    @PutMapping("/project/{projectid}/gatewayinfo/{resource_id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public ResponseId updateGatewayInfoForZeta(@PathVariable("projectid") String projectId, @PathVariable("resource_id") String vpcId, @RequestBody GatewayInfoJson gatewayInfoJson) throws Exception {
```

Comment:
Who will call this interface? vpc-manager?

Reply:
Not sure, now it’s just providing this interface

Reply:
DPM will call this interface if there is no gatewayinfo entry in the DPM's cache for a vpc.